src/Swarm/Language/Syntax.hs

```diff
@@ -960,6 +967,7 @@ erase (SApp s1 s2) = TApp (eraseS s1) (eraseS s2)
 erase (SLet r x mty s1 s2) = TLet r (lvVar x) mty (eraseS s1) (eraseS s2)
 erase (SDef r x mty s) = TDef r (lvVar x) mty (eraseS s)
 erase (SBind mx s1 s2) = TBind (lvVar <$> mx) (eraseS s1) (eraseS s2)
+erase (SAnnotate s _) = eraseS s
```

Comment:
Oh, I didn't notice this before, we should preserve the `SAnnotate` constructor here too, for the same reasons as elsewhere.  Erasing types just means erasing the inferred types the typechecker filled in, not erasing an explicit type annotation the user wrote.